tests/check_non_terminating.sh

```diff
@@ -0,0 +1,166 @@
+    devfilePath=$1
+    local _gfcca_args=()
+    local _gfcca_args_string=()
+
+    IFS=" " read -r -a _gfcca_args_string <<< "$($YQ_PATH eval '[ .components[] | select(has("container")) ] | .[0].container.args[]? + " "' "$devfilePath" -r | paste -s -d '\0' -)"
+    if (( ${#_gfcca_args_string[@]} == 0 )); then
+      echo ""
+    else
+        for arg in "${_gfcca_args_string[@]}"; do 
+            _gfcca_args+=("${arg}")
+        done
+        echo "${_gfcca_args[@]}"
+    fi
+}
+
+isNonTerminating() {
+    _int_image=$1
+    _int_command=("$2")
+    _int_command_args=("$3")
+
+    namespace=default
+    timeout_in_sec=60 # <== includes image pulling
+
+    echo "  PARAMS: image --> $_int_image, command --> ${_int_command[*]}, args --> ${_int_command_args[*]}"
+ 
+    if [ "${_int_command[*]}" == "null" ] && [ "${_int_command_args[*]}" == "null" ]; then
+        echo "  COMMAND: \"kubectl run test-terminating -n default --attach=false --restart=Never --image=$_int_image\""
+        2>/dev/null 1>/dev/null kubectl run test-terminating -n default --attach=false --restart=Never --image="$_int_image"
+    elif [ "${_int_command[*]}" == "null" ]; then
+        echo "  COMMAND: \"kubectl run test-terminating -n default --attach=false --restart=Never --image=$_int_image -- ${_int_command_args[*]}\""
+        2>/dev/null 1>/dev/null kubectl run test-terminating -n default --attach=false --restart=Never --image="$_int_image" -- ${_int_command_args[*]}
+    elif [ "${_int_command_args[*]}" == "null" ]; then
+        echo "  COMMAND: \"kubectl run test-terminating -n default --attach=false --restart=Never --image=$_int_image --command -- ${_int_command[*]}\""
+        2>/dev/null 1>/dev/null kubectl run test-terminating -n default --attach=false --restart=Never --image="$_int_image" --command=true -- ${_int_command[*]}
+    else
+        echo "  COMMAND: \"kubectl run test-terminating -n default --attach=false --restart=Never --image=$_int_image --command -- ${_int_command[*]} ${_int_command_args[*]}\""
+        2>/dev/null 1>/dev/null kubectl run test-terminating -n default --attach=false --restart=Never --image="$_int_image" --command=true -- ${_int_command[*]} ${_int_command_args[*]}
+    fi
+    
+    if 2>/dev/null 1>/dev/null kubectl wait pods -n ${namespace} test-terminating --for condition=Ready --timeout=${timeout_in_sec}s; then
+      echo "  SUCCESS: The container started successfully and didn't terminate"
+      2>/dev/null 1>/dev/null kubectl delete pod --force test-terminating -n default
+      return 0
+    else
+      echo "  ERROR: Failed to reach \"Ready\" condition after $timeout_in_sec seconds"
+      echo "  ↓↓↓↓↓↓↓↓↓ Pod description ↓↓↓↓↓↓↓↓"
+      echo ""
+      kubectl describe pod -n ${namespace} test-terminating
+      echo ""
+      echo "  ↑↑↑↑↑↑↑↑↑ Pod description ↑↑↑↑↑↑↑↑"
+      2>/dev/null 1>/dev/null kubectl delete pod test-terminating -n default
+      return 1
+    fi
+}
+
+YQ_PATH=yq
```

Comment:
Not sure if this is left over from an earlier refactor -- should this be something like

```suggestion
YQ_PATH=${YQ_PATH:-yq}
```

Reply:
Also this would be very useful to have as this script uses `mikefarah/yq` whereas my default installed binary is the `kislyuk/yq` version. To use the script I would need to `export YQ_PATH=...`.

Reply:
👍